### PR TITLE
PP-5486: Use the system proxy settings for the Jersey client

### PR DESCRIPTION
**Pull request by rhowe-gds**

### CODE REVIEW

File: src/main/java/uk/gov/pay/directdebit/notifications/clients/ClientFactory.java [hunk 1]
```diff
@@ -20,9 +21,9 @@ public ClientFactory(Environment environment, DirectDebitConfig conf) {
     }
 
     public Client createWithDropwizardClient(String name) {
-
         Client client = new JerseyClientBuilder(environment)
                 .using(conf.getClientConfiguration())
+                .using(new SystemDefaultRoutePlanner(null))
```
Comment (oswaldquek) [inline src/main/java/uk/gov/pay/directdebit/notifications/clients/ClientFactory.java]: why is this needed? connector doesn't seem to use it...

Comment (RoryMalcolm) [reply src/main/java/uk/gov/pay/directdebit/notifications/clients/ClientFactory.java]: Because connector does not speak to external services and internal services like adminusers, DD connector is in a special situation where its client needs to have both of these functionalities and therefore needs special allowances with regards to nonProxyHosts
